pythonforandroid/toolchain.py

```diff
@@ -792,9 +793,9 @@ def export_dist(self, args):
 
     @property
     def _dist(self):
-        ctx = self.ctx
-        dist = dist_from_args(ctx, self.args)
-        return dist
+        if (not hasattr(self, 'dist') or self.dist is None):
+            self.dist = dist_from_args(self.ctx, self.args)
+        return self.dist
```

Comment:
I have to say this feels unnatural somehow:
1) I would simplify the initialisation of the attribute by assigning it to `None` within the constructor
2) It feels weird to have the property being "self._protected" while the member "self.public". Maybe we want to call the member  `__dist` at least? I don't know